storage/src/adb/any/sync/client.rs

```diff
@@ -195,11 +196,38 @@ where
                 buffer_pool: config.db_config.buffer_pool.clone(),
             },
             config.lower_bound_ops,
+            config.upper_bound_ops,
         )
         .await
         .map_err(adb::Error::JournalError)
         .map_err(Error::Adb)?;
 
+        // Check how many operations are already in the log.
+        let log_size = log
+            .size()
+            .await
+            .map_err(|e| Error::Adb(adb::Error::JournalError(e)))?;
+
+        if log_size == config.upper_bound_ops + 1 {
```

Comment:
Are we ok with log_size >= config.upper_bound_ops + 1 here? Or should that be handled elsewhere (if so, I suggest assertion)?

Reply:
I added an assertion in the fixed journal's `init_sync`: 

```rs
        assert!(journal_size <= upper_bound + 1);
        assert!(journal_size >= lower_bound);
```

The returned log should never have a size above `upper_bound + 1`.

Reply:
Could you leave an explicit comment above 211 saying this is assumed? I think it'll help anyone reviewing this code in the future.

Reply:
I also think its worth asserting journal_size <= upper_bound + 1 here too.